### PR TITLE
createSymLinks.pl rewritten in bash

### PR DESCRIPTION
**Pull request by tulamor**
The script is still in progress

### CODE REVIEW

File: SCRAM/createSymLinks.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    if [[ $arg =~ ^-(.+)$ ]]; then
+      srcnfilter="${srcnfilter}${BASH_REMATCH[1]}|"
+    elif [[ $arg =~ ^(\+|)(.+)$ ]]; then
+      srcfilter="${srcfilter}${BASH_REMATCH[2]}|"
+    fi
+    shift
+  done
+fi
+srcfilter="$(echo "$srcfilter" | sed 's/\|$//')"
+srcnfilter="$(echo "$srcnfilter" | sed 's/\|$//')"
```
Comment (smuzaffar) [inline SCRAM/createSymLinks.sh:25]: change `sed 's/\|$//'` to  `sed 's/|$//'` in above two lines

File: SCRAM/createSymLinks.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+function getSubDir () {
+  local dir=$1
+  local sdir=$2
+  if [ "X$sdir" = "X" ] ; then printf "\n" && return; fi
+  if [ "X$sdir" = "X." ] ; then sdir=$"/`basename $dir`" ; else sdir="/${sdir}" ; fi
```
Comment (smuzaffar) [inline SCRAM/createSymLinks.sh:31]: typo here. change `sdir=$"/`basename $dir`"` to `sdir=$"/`basename $sdir`"`

File: SCRAM/createSymLinks.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  printf "$sdir\n"
+}
+
+if [ -d "$src" ] ; then
+  for dir in "$(find $src -maxdepth $depth -mindepth $depth -name "*" -type d)"; do
```
Comment (smuzaffar) [inline SCRAM/createSymLinks.sh:36]: remove double quotes here otherwise you only get one item.

File: SCRAM/createSymLinks.sh [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    rpath="$(echo "$rpath" | sed "s|${src}/*||")"
+    sdir=$(getSubDir "$dir" "$subdir")
+    ldir=$(getSubDir "$dir" "$linkdir")
+    slink="${des}/${rpath}${ldir}"
+    printf "$slink \n"
+    if [ -d "${dir}${sdir}" ]; then
+      if [ "$des" = python ] &&  [ -d $slink ]; then rm -rf $slink; fi
+      slinkdir=$(dirname "$slink")
+      ldir="slinkdir"
+      ldir="$(echo "$ldir" | sed 's/[a-zA-Z0-9-_]+/../g')"
```
Comment (smuzaffar) [inline SCRAM/createSymLinks.sh:50]: I think sed here is not working properly

File: SCRAM/createSymLinks.sh (hunk 2 above)
Comment (tulamor) [reply SCRAM/createSymLinks.sh:31]: > typo here. change `sdir=$"/`basename $dir`"` to `sdir=$"/`basename $sdir`"`

Perl version:
  if ($sdir eq "."){$sdir="/".basename($dir);}
  else{$sdir="/${sdir}";}

Should it be: basename $sdir or leave as in perl version?

Comment (smuzaffar) [reply SCRAM/createSymLinks.sh:31]: you are right, I did not see the `dir` variable at line number 28

File: SCRAM/createSymLinks.sh [hunk 5]
```diff
@@ -61,7 +60,7 @@ if [ -d "$des" ] ; then
  declare -A rm
  declare -A ok
  for d in $(find $des -name "*" -type l) ; do
-  local d1 = d
+  d1=$d
   d1="$(echo "d1" | sed 's/\/[^\/]+$//')"
```
Comment (smuzaffar) [inline SCRAM/createSymLinks.sh:64]: why `echo "d1"` and not `echo "$d1"`

**Pull request merged**